databricks/koalas/series.py

```diff
@@ -1209,8 +1226,55 @@ def fillna(self, value=None, axis=None, inplace=False):
         4    0.0
         5    6.0
         Name: x, dtype: float64
+
+        We can also propagate non-null values forward or backward.
+
+        >>> s.fillna(method='ffill')
+        0    NaN
+        1    2.0
+        2    3.0
+        3    4.0
+        4    4.0
+        5    6.0
+        Name: x, dtype: float64
         """
-        kseries = _col(self.to_dataframe().fillna(value=value, axis=axis, inplace=False))
+        if axis is None:
+            axis = 0
+        if not (axis == 0 or axis == "index"):
+            raise NotImplementedError("fillna currently only works for axis=0 or axis='index'")
+        if (value is None) and (method is None):
+            raise ValueError("Must specify a fill 'value' or 'method'.")
+
+        column_name = self.name
+        scol = self._scol
+        if value is not None:
+            if not isinstance(value, (float, int, str, bool)):
+                raise TypeError("Unsupported type %s" % type(value))
+            if limit is not None:
+                raise ValueError('limit parameter for value is not support now')
+            scol = F.when(scol.isNull(), value).otherwise(scol)
+        else:
+            if method in ['ffill', 'pad']:
+                func = F.last
+                end = (Window.currentRow - 1)
+                if limit is not None:
+                    begin = Window.currentRow - limit
+                else:
+                    begin = Window.unboundedPreceding
+            elif method in ['bfill', 'backfill']:
+                func = F.first
+                begin = Window.currentRow + 1
+                if limit is not None:
+                    end = Window.currentRow + limit
+                else:
+                    end = Window.unboundedFollowing
+            else:
+                raise ValueError('Expecting pad, ffill, backfill or bfill.')
+            window = Window.orderBy(self._internal.index_columns).rowsBetween(begin, end)
+            scol = F.when(scol.isNull(), func(scol, True).over(window)).otherwise(scol)
+
+        kseries = Series(self._kdf._internal.copy(scol=scol), anchor=self._kdf)\
+            .rename(column_name).astype(np.float64)
```

Comment:
I don't think we can simply cast to `float64`here. If the integer series doesn't have null values, it preserves the integer type:

```py
>>> pd.Series([1,2,3,4]).fillna(method='ffill')
0    1
1    2
2    3
3    4
dtype: int64
```